File: fix-discord-links.js
const fs = require('fs');
const path = require('path');

const CORRECT_DISCORD = 'https://discord.gg/ring-0';
const WRONG_PATTERNS = [
  'https://discord.gg/skylineggs',
  'https://discord.gg/skylinecheats',
  'https://discord.gg/ring-0',
  'discord.gg/skylineggs',
  'discord.gg/skylinecheats',
  'discord.gg/ring-0'
];

function fixDiscordLinks(dir) {
  const files = fs.readdirSync(dir);
  let fixedCount = 0;

  files.forEach(file => {
    const filePath = path.join(dir, file);
    const stat = fs.statSync(filePath);

    // Skip node_modules, .next, .git
    if (file === 'node_modules' || file === '.next' || file === '.git') {
      return;
    }

    if (stat.isDirectory()) {
      fixedCount += fixDiscordLinks(filePath);
    } else if (stat.isFile()) {
      // Process text files
      const ext = path.extname(file);
      if (['.tsx', '.ts', '.jsx', '.js', '.md', '.html', '.json', '.sql', '.txt'].includes(ext)) {
        try {
          let content = fs.readFileSync(filePath, 'utf8');
          let modified = false;

          WRONG_PATTERNS.forEach(pattern => {
            if (content.includes(pattern)) {
              const regex = new RegExp(pattern.replace(/[.*+?^${}()|[\]\\]/g, '\\$&'), 'g');
              content = content.replace(regex, CORRECT_DISCORD);
              modified = true;
            }
          });

          if (modified) {
            fs.writeFileSync(filePath, content, 'utf8');
            console.log(`✓ Fixed: ${filePath}`);
            fixedCount++;
          }
        } catch (err) {
          // Skip files that can't be read as text
        }
      }
    }
  });

  return fixedCount;
}

console.log('Fixing Discord invite links to: ' + CORRECT_DISCORD);
console.log('Starting...\n');

const fixed = fixDiscordLinks('.');
console.log(`\n✓ Complete! Fixed ${fixed} files`);

File: fix-discord-links.py
import os
import re

def replace_discord_links(directory):
    """Replace discord.gg/skylineggsu with discord.gg/skylineggs in all files"""
    
    # File extensions to process
    extensions = ['.tsx', '.ts', '.js', '.jsx', '.md']
    
    files_updated = 0
    total_replacements = 0
    
    for root, dirs, files in os.walk(directory):
        # Skip node_modules and .next directories
        if 'node_modules' in root or '.next' in root or '.git' in root:
            continue
            
        for file in files:
            if any(file.endswith(ext) for ext in extensions):
                file_path = os.path.join(root, file)
                
                try:
                    with open(file_path, 'r', encoding='utf-8') as f:
                        content = f.read()
                    
                    # Replace the Discord link
                    new_content = content.replace('discord.gg/skylineggsu', 'discord.gg/skylineggs')
                    
                    # Count replacements in this file
                    replacements = content.count('discord.gg/skylineggsu')
                    
                    if replacements > 0:
                        with open(file_path, 'w', encoding='utf-8') as f:
                            f.write(new_content)
                        
                        files_updated += 1
                        total_replacements += replacements
                        print(f"✅ Updated {file_path} ({replacements} replacements)")
                
                except Exception as e:
                    print(f"❌ Error processing {file_path}: {e}")
    
    print(f"\n🎉 Complete!")
    print(f"Files updated: {files_updated}")
    print(f"Total replacements: {total_replacements}")

if __name__ == "__main__":
    # Run from the magma src directory
    replace_discord_links(".")
    print("\n✅ All Discord links updated to discord.gg/skylineggs")
